### DISCUSSION TITLE
Passing the params from a component file to a dynamic route

### DISCUSSION THREAD

**Discussion by useDeep** [vercel/next.js#52954] [App Router]
So,,,,, I have a search component rendered on the app/page.js file. When the user types in the search box, I am storing in a state. I also have a app/search/[searchItem]/page.jsx file. Now when the user enters the search item they are looking for, I want the app to dynamically route to the mentioned page, which takes the searchItem as the params.

I have tried using `useRoute` hook and `route.push(`/search/${searchItem}`)`, but it doesn't seem to work.
I have also tried `redirect(`/search/${searchTerm}`)`.

I guess is that the page is re rendering and losing the searchItem state. 

Appreciate your help on fixing this. Here is that search [component](https://github.com/useDeep/youtube-clone/blob/main/components/search/SearchBar.jsx).

**Comment by devjiwonchoi** [comment]
Simple answer: add an `action` option for the redirect URL.

```js
action={`/search/${searchTerm}`}
```

Additionally, I'd recommend you to look up `useSearchParams` and `searchParams` for Dynamic Routes.

Note: Look for [useSearchParams Behavior](https://nextjs.org/docs/app/api-reference/functions/use-search-params#behavior) and config Rendering option based on your needs.

### Reference
- [useSearchParams](https://nextjs.org/docs/app/api-reference/functions/use-search-params)
- [searchParams](https://nextjs.org/docs/app/api-reference/file-conventions/page#searchparams-optional)
